corehq/apps/userreports/tasks.py

```diff
@@ -303,119 +304,139 @@ def _queue_chunk(indicators):
+    ]
     with CriticalSection(lock_keys):
-        indicators = AsyncIndicator.objects.filter(doc_id__in=doc_ids)
-        if not indicators:
+        all_indicators = AsyncIndicator.objects.filter(
+            doc_id__in=indicator_doc_ids
+        )
+        if not all_indicators:
             return
 
-        first_indicator = indicators[0]
-        processed_indicators = []
-        failed_indicators = []
-
-        for i in indicators:
-            assert i.domain == first_indicator.domain
-            assert i.doc_type == first_indicator.doc_type
-
-        indicator_by_doc_id = {i.doc_id: i for i in indicators}
-        doc_store = get_document_store_for_doc_type(first_indicator.domain, first_indicator.doc_type)
-        indicator_config_ids = first_indicator.indicator_config_ids
-        related_docs_to_rebuild = set()
+        doc_store = get_document_store_for_doc_type(
+            all_indicators[0].domain, all_indicators[0].doc_type
+        )
+        related_doc_ids = set()
+        failed_indicators = set()
 
+        rows_to_save_by_adapter = defaultdict(list)
+        doc_ids_to_save_by_adapter = defaultdict(list)
+        indicator_by_doc_id = {i.doc_id: i for i in all_indicators}
+        config_ids = set()
         with timer:
             for doc in doc_store.iter_documents(list(indicator_by_doc_id.keys())):
                 indicator = indicator_by_doc_id[doc['_id']]
-                successfully_processed, to_remove, rebuild_related_docs = _save_document_helper(indicator, doc)
-                if rebuild_related_docs:
-                    related_docs_to_rebuild = related_docs_to_rebuild.union(icds_get_related_docs_ids(doc['_id']))
-                if successfully_processed:
-                    processed_indicators.append(indicator.pk)
+                eval_context = EvaluationContext(doc)
+                for config_id in indicator.indicator_config_ids:
+                    config_ids.add(config_id)
+                    try:
+                        config = _get_config(config_id)
+                    except (ResourceNotFound, StaticDataSourceConfigurationNotFoundError):
+                        celery_task_logger.info("{} no longer exists, skipping".format(config_id))
+                        # remove because the config no longer exists
+                        _mark_config_to_remove(config_id, [indicator.pk])
+                        continue
+                    except ESError:
+                        celery_task_logger.info("ES errored when trying to retrieve config")
+                        failed_indicators.add(indicator)
+                        continue
+                    adapter = None
+                    try:
+                        adapter = get_indicator_adapter(config, can_handle_laboratory=True)
+                        rows_to_save_by_adapter[adapter].extend(adapter.get_all_values(doc, eval_context))
+                        doc_ids_to_save_by_adapter[adapter].append(doc['_id'])
```

Comment:
I think you should be able to remove `doc_ids_to_save_by_adapter` because you can get the same info by using `{row['doc_id'] for row in rows_to_save_by_adapter[adapter]`} later

Reply:
Okay, I thought there might be a case where rows may not carry this from the existing logic in `_save_rows` which takes the doc_id explicitly as an argument. If, as you say that's not true, we don't even need to pass the doc_id to `_save_rows`, so I will get rid of the extra argument.